synfig-studio/src/gui/widgets/widget_timetrack.cpp

```diff
@@ -198,6 +198,44 @@ bool Widget_Timetrack::move_selected(synfig::Time delta_time)
 	return ok;
 }
 
+ void Widget_Timetrack::interpolate_selected(synfig::Interpolation type)
+{
+	synfig::Waypoint::Model model;
+	model.set_before(type);
+	model.set_after(type);
+	std::vector<WaypointItem*> selection = waypoint_sd.get_selected_items(); //storing selected waypoint items
+
+	for(int i =0 ; i<selection.size() ; i++) //iteratre through selected waypoint items to fetch the set and apply the interpolation type
+    {
+		std::set<synfig::Waypoint, std::less<synfig::UniqueID> > waypoint_set_new;
+		fetch_waypoints(*selection[i], waypoint_set_new);
+		std::set<synfig::Waypoint, std::less<synfig::UniqueID> >::const_iterator iter;
+
+		for(iter=waypoint_set_new.begin();iter!=waypoint_set_new.end();++iter)
+        {
+			synfigapp::Action::PassiveGrouper group(get_canvas_interface()->get_instance().get(),_("Change Waypoint Group"));
```

Comment:
1. I think you don't need the `.get()` of `get_canvas_interface()->get_instance().get()`
2. The intention of PassiveGrouper is to group several actions inside a single one: with a single Undo, all inside actions are undone at once. So it does not make sense to make several PassiveGroups in a row, each one containing a single action. Place it outside the loops.

Reply:
1 - i wonder why ? 
     removing gives out the error
   ` error: no matching function for call to 'synfigapp::Action::PassiveGrouper::PassiveGrouper(const etl::loose_handle<synfigapp::Instance>&, char*)'
  208 |         synfigapp::Action::PassiveGrouper group(get_canvas_interface()->get_instance(),_("Change Waypoint Group"));`
2- definetly makes sense, thanks for notifying.